lib/Sema/CSSimplify.cpp

```diff
@@ -2040,6 +2040,19 @@ bool ConstraintSystem::repairFailures(
     return false;
   };
 
+  auto repairByAddingConformance = [&](Type lhs, Type rhs) -> bool {
+    if (lhs->isTypeVariableOrMember())
```

Comment:
What happens if you don’t bail out in this case?

Reply:
I think it's going to clash with Existential restriction, but I'm still working out all of the logic here...